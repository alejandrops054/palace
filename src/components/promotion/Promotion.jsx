/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import {React, useState} from 'react';
import {Input, Select, Checkbox} from "../index";
import Modal from 'react-bootstrap/Modal'
import TextField from '@material-ui/core/TextField';
import Button from 'react-bootstrap/Button'
import TablaPromotion from './componets/TablaPromotion'

const Promotions = ()=>{

    const Titulo =[
        'Promotions'
    ]

    const [show, setShow] = useState(false);

    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);

    return (
        <>
        <Button variant="primary" onClick={handleShow}>
          {Titulo}
        </Button>
  
        <Modal show={show} onHide={handleClose} size="lg" aria-labelledby="contained-modal-title-vcenter" centered>
          <Modal.Header closeButton>
            <Modal.Title>{Titulo}</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <div className="container">
                <div className="row">
                  <div className="col-md-12">
                    <TablaPromotion/>
                  </div>
                </div>
            </div>
          </Modal.Body>
          <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
              Cancel
            </Button>
          </Modal.Footer>
        </Modal>
        </>
    )
}

export default Promotions