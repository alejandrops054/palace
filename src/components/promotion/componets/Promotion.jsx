/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import React from "react";
import {Input, Select, Checkbox} from "../../index";

export default function ValoresPromotion(){
    return (
        <>
        <from>
            <div className="container">
                <div className="row">
                    <div className="col-md-4">
                        <Input type="text" col={12} label="ID"/>
                        <Checkbox col={12} label="Closed Date"/>
                        <Checkbox col={12} label="Unlimited"/>
                        <Checkbox col={12} label="Trasfenble"/>
                    </div>
                    <div className="col-md-4">
                        <Select option="Alta" col={12} label="Status"/>
                    </div>
                    <div className="col-md-4">
                        <Select option="Promotion Type" col={12} label="Referrals Plus"/>
                    </div>
                </div>
                <br/>
                <div className="row">
                    <div className="col-md-4">
                        <Input type="text" col={12} label="Spanish Description"/>
                        
                    </div>
                    <div className="col-md-4">
                        <Input type="text" col={12} label="Short Description MX"/>
                    </div>
                    <div className="col-md-4">  
                        <br/>
                        <Checkbox col={12} label="Retricted"/>
                       
                    </div>
                    <div className="col-md-4">
                        <Input type="text" col={12} label="English Description"/>
                        <Input type="text" col={12} label="Benefeit Type"/>
                    </div>
                    <div className="col-md-4">
                        <Input type="text" col={12} label="Short Description US"/>
                        <Input label="Date from" type="date" placeholder="" col={12}/>
                    </div>
                    <div className="col-md-4">
                        <Select option="0" type="text" col={12} label="Max Use"/><br/>
                        <Input label="Date To" type="date" placeholder="" col={12}/>
                    </div>
                    <div className="col-md-2">
                        <Select option="0" type="text" col={12} label="Order"/>
                    </div>
                    <div className="col-md-2">
                        <Select option="0" type="text" col={12} label="Validity"/>
                    </div>
                    <div className="col-md-2">
                        <Select option="0" type="text" col={12} label="Year from"/>
                    </div>
                    <div className="col-md-2">
                        <Select option="0" type="text" col={12} label="Year To"/>
                    </div>
                    <div className="col-md-2">
                        <Select option="0" type="text" col={12} label="Couples"/>
                    </div>
                    <div className="col-md-2">
                        <Select option="0" type="text" col={12} label="Weeks"/>
                    </div>
                </div>
                <br/>
            </div>
        </from>
        </>
    )
}