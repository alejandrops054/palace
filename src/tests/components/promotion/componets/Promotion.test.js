/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Promotion from '../../../../components/promotion/componets/Promotion'

describe('Promotion component',() => {
    test('Debe mostrar snapshot', () => {
        
        const wrapper = shallow(<Promotion/>)
        expect(wrapper).toMatchSnapshot();
    })
    
})