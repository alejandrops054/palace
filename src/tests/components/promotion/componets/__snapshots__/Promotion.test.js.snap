// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promotion component Debe mostrar snapshot 1`] = `
<Fragment>
  <from>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <Input
            col={12}
            label="ID"
            type="text"
          />
          <Checkbox
            col={12}
            label="Closed Date"
          />
          <Checkbox
            col={12}
            label="Unlimited"
          />
          <Checkbox
            col={12}
            label="Trasfenble"
          />
        </div>
        <div
          className="col-md-4"
        >
          <Select
            col={12}
            label="Status"
            option="Alta"
          />
        </div>
        <div
          className="col-md-4"
        >
          <Select
            col={12}
            label="Referrals Plus"
            option="Promotion Type"
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <Input
            col={12}
            label="Spanish Description"
            type="text"
          />
        </div>
        <div
          className="col-md-4"
        >
          <Input
            col={12}
            label="Short Description MX"
            type="text"
          />
        </div>
        <div
          className="col-md-4"
        >
          <br />
          <Checkbox
            col={12}
            label="Retricted"
          />
        </div>
        <div
          className="col-md-4"
        >
          <Input
            col={12}
            label="English Description"
            type="text"
          />
          <Input
            col={12}
            label="Benefeit Type"
            type="text"
          />
        </div>
        <div
          className="col-md-4"
        >
          <Input
            col={12}
            label="Short Description US"
            type="text"
          />
          <Input
            col={12}
            label="Date from"
            placeholder=""
            type="date"
          />
        </div>
        <div
          className="col-md-4"
        >
          <Select
            col={12}
            label="Max Use"
            option="0"
            type="text"
          />
          <br />
          <Input
            col={12}
            label="Date To"
            placeholder=""
            type="date"
          />
        </div>
        <div
          className="col-md-2"
        >
          <Select
            col={12}
            label="Order"
            option="0"
            type="text"
          />
        </div>
        <div
          className="col-md-2"
        >
          <Select
            col={12}
            label="Validity"
            option="0"
            type="text"
          />
        </div>
        <div
          className="col-md-2"
        >
          <Select
            col={12}
            label="Year from"
            option="0"
            type="text"
          />
        </div>
        <div
          className="col-md-2"
        >
          <Select
            col={12}
            label="Year To"
            option="0"
            type="text"
          />
        </div>
        <div
          className="col-md-2"
        >
          <Select
            col={12}
            label="Couples"
            option="0"
            type="text"
          />
        </div>
        <div
          className="col-md-2"
        >
          <Select
            col={12}
            label="Weeks"
            option="0"
            type="text"
          />
        </div>
      </div>
      <br />
    </div>
  </from>
</Fragment>
`;
